### PR TITLE
test: add test for helm template rendering

### PR DESCRIPTION
**Pull request by akshaysngupta**
## Checklist
- [x] The title of the PR is clear and informative
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

This PR adds test for helm template rendering for correctness.

Second commit contains a test for:
* tolerations
* nodeSelector
* affinity

image

### CODE REVIEW

File: helm/ingress-azure/tests/snapshots.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	}
+
+	for expFile := range expected.files {
+		if _, ok := s.files[expFile]; !ok {
+			t.Errorf("expected snapshot has file missing in actual: %s", expFile)
+		}
+	}
+}
+
+// RenderChart renderss a Helm chart to a given directory.
```
Comment (3quanfeng) [inline helm/ingress-azure/tests/snapshots.go:52]: typo "renderss"

File: helm/ingress-azure/tests/chart_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+const valuesDir = "fixtures"
+
+// TestChart is a simple snapshot-style regression test for the CCP chart.
```
Comment (3quanfeng) [inline helm/ingress-azure/tests/chart_test.go:20]: not specific for CCP chart?

Comment (akshaysngupta) [reply helm/ingress-azure/tests/chart_test.go:20]: whoops... missed that

**Review by 3quanfeng — DISMISSED** [review]
LGTM

**Review by 3quanfeng — APPROVED** [review]

**Pull request merged**